### ISSUE TITLE
Can't find variable: _setGlobalConsole [Typescript/React Navigation preset]

### ISSUE THREAD

**Issue by juancebarberis** [software-mansion/react-native-reanimated#1869]
## Description

On a clean Expo project, using the Typescript w/ React Navigation preset, I'm getting this error on both devices, Android and iOS.

I search through previously issues but all the 'trivial' solutions (like cache, reinstall, etc) didn't work for me. 

I've also configure my babel.config.js 
`
module.exports = function(api) {
  api.cache(true);
  return {
    presets: [
      'babel-preset-expo'
    ],
    plugins: [
      'react-native-reanimated/plugin'
    ]
  };
};
`

![iOS screenshot](https://i.imgur.com/w7IGUsr.jpg)

![Android screenshot](https://i.imgur.com/QZsC2yZ.jpg)

On iOS, you can skip the error. Android doesn't not allow you to use the app.

### Expected behavior

App run without this error.

### Actual behavior 

On app startup, it shows this error.

### Steps to reproduce

1. Create a new React Native app with expo. 
2. Select Typescript preset with React Navigation (tabs example).
3. Open it with expo on any device.

## Snack or minimal code example

No code to show, it's a basci pre-generated project. 

## Package versions

- React Native: https://github.com/expo/react-native/archive/sdk-40.0.1.tar.gz
- React Native Reanimated: 2.0.0
- NodeJS: 12.18.3
- Xcode: -
- Java & Gradle: -

## Affected platforms

- [ ] Android
- [ ] iOS

**Comment by juancebarberis** [comment]
So... after diggin deeper on other issues, I ended up touching the `/my-app/node_modules/react-native-reanimated/src/reanimated2/core.js` file, and commented the line wich calls the undefined _setGlobalConsole(console). It works for the moment I needed the project running on the client device, but of course it is an horrible solution.

**Comment by AsenCME** [comment]
I am experiencing the same issue. I originally believed it was caused by other libraries but now I am almost fully convinced that it is just a matter of incompatibility between react-native-reanimated@2.0.0 and Expo. Thanks for the workaround @juancebarberis

**Comment by piaskowyk** [comment]
It should be fixed with `2.0.1`
https://github.com/software-mansion/react-native-reanimated/blob/a8d9720ee5312a43cc4e3beb02e70b2fc6a030ae/src/reanimated2/core.ts#L12-L17

**Comment by AbhijeetGavali** [comment]
I got this error today while running my new app

```
ERROR  ReferenceError: Property '_setGlobalConsole' doesn't exist

This error is located at:
    in App
    in RCTView (created by View)
    in View (created by AppContainer)
    in RCTView (created by View)
    in View (created by AppContainer)
    in AppContainer
    in sknisb(RootComponent), js engine: hermes
```

![image](https://github.com/software-mansion/react-native-reanimated/assets/81749367/0f788dde-eb70-405b-99e2-7faf73678043)

below is my dependancy list of project
```
"dependencies": {
    "react": "18.2.0",
    "react-native": "0.72.1",
    "react-native-reanimated": "^3.3.0",
    "react-native-vision-camera": "^2.15.4",
    "vision-camera-code-scanner": "^0.2.0"
  },
  "devDependencies": {
    "@babel/core": "^7.20.0",
    "@babel/preset-env": "^7.20.0",
    "@babel/runtime": "^7.20.0",
    "@react-native/eslint-config": "^0.72.2",
    "@react-native/metro-config": "^0.72.7",
    "@tsconfig/react-native": "^3.0.0",
    "@types/metro-config": "^0.76.3",
    "@types/react": "^18.0.24",
    "@types/react-test-renderer": "^18.0.0",
    "babel-jest": "^29.2.1",
    "eslint": "^8.19.0",
    "jest": "^29.2.1",
    "metro-react-native-babel-preset": "0.76.5",
    "prettier": "^2.4.1",
    "react-test-renderer": "18.2.0",
    "typescript": "4.8.4"
  },
```
